carma-messenger-core/truck_inspection_plugin/test/test_truck_inspection_plugin.cpp

```diff
@@ -1,148 +0,0 @@
-    std::string weight_= "15654";
-    std::string iss_score_ = "75";
-    std::string permit_required_= "yes";
-    std::string ads_health_status = "engaged"; //ads_health_status
-    std::string ads_auto_status_ = "1"; //CAUTION
-    
-    cav_msgs::MobilityOperation msg_out;
-    cav_msgs::MobilityHeader msg_out_header;
-    msg_out.strategy = "TruckInspection";
-    msg_out.strategy_params = "'vin_number:"+vin_number_
-                              +",carrier_name:"+carrier_name_
-                              +",date_of_last_state_inspection:"+date_of_last_state_inspection_
-                              +",date_of_last_ads_calibration:"+date_of_last_ads_calibration_
-                              +",license_plate:"+license_plate_
-                              +",weight:"+weight_
-                              +",carrier_id:"+carrier_id_
-                              +",permit_required:"+permit_required_
-                              +",iss_score:"+iss_score_
-                              +",pre_trip_ads_health_check:"+pre_trip_ads_health_check_
-                              +",ads_health_status:"+ads_health_status_
-                              +",ads_auto_status:"+ads_auto_status_
-                              +",ads_software_version:"+ads_software_version_
-                              +",state_short_name:"+state_short_name_+"'";
-    msg_out_header.sender_id =  "''";
-    msg_out_header.recipient_id="''";
-    msg_out_header.sender_bsm_id="'5055'";
-    msg_out_header.plan_id= "";
-    msg_out_header.timestamp= 1586291827962;
-    msg_out.m_header = msg_out_header;
-
-    mobility_operation_inbound_pub.publish(msg_out);
-    std::this_thread::sleep_for(std::chrono::milliseconds(500));
-
-    // Spin so that publication can get to subscription
-    ros::spinOnce(); 
-   
-    //ASSERTION
-    /**Require running: roslaunch truck_inspection_plugin test_launch.test and rosrun truck_inspection_plugin truck_inspection_plugin_test 
-     * Before changing 0 to 1 making sure there is a subscriber to this topic***/
-    EXPECT_EQ(1,mobility_operation_inbound_pub.getNumSubscribers());
-}
-
-
-TEST(TruckInspectionTest,TestTruckIdentified){
-    ros::NodeHandle nh = ros::NodeHandle();
-    TestSubscriber subscriber_Iden;
-
-    // Subscriber
-    ros::Subscriber cav_truck_identified_sub = nh.subscribe("cav_truck_identified", 5,&TestSubscriber::callback, &subscriber_Iden); 
-    std::this_thread::sleep_for(std::chrono::milliseconds(500));
-    
-    // Spin so that publication can get to subscription
-    ros::spinOnce(); 
-
-    //ASSERTION
-    EXPECT_EQ(1, cav_truck_identified_sub.getNumPublishers()); 
-    
-    //Require topic mobility_operation_inbound publishes data (cav_truck_identified will then publish the data) before run below test
-    if(subscriber_Iden.receivedMessage)
-        EXPECT_EQ("vin_number:1FUJGHDV0CLBP8896,license_plate:DOT-1003,state_short_name:VA",subscriber_Iden.message->data);
```

Comment:
This check is missing from the ROS2 tests and needs to be included as its the most important check in all the tests since it actually verifies message contents.